packages/project/src/merge/map-uuids.ts

```diff
@@ -39,82 +40,234 @@ export interface MappingResults {
 }
 
 export default (source: Workflow, target: Workflow): MappingResults => {
-  const nodeMapping: Record<string, MappingRule> = {};
   const edgeMapping: Record<string, MappingRule> = {};
-  // const mapping: Record<string, MappingRule> = {};
 
-  // sets all nodes as not existing in target
-  const targetIds: Record<string, string> = {};
-  for (const tstep of target.steps) {
-    nodeMapping[tstep.id] = null;
-    targetIds[tstep.id] = tstep.openfn?.uuid || tstep.id;
+  // Map by id
+  let {
+    mapping: nodeMapping,
+    pool,
```

Comment:
the way the pool is handled here - splitting into nodes and edges - makes me think my algorithm isn't working. This is important!

It's totally fine if the algorithm doesn't work, but I need to know so that we can fix it and we can share it with the other team